test/test_atacac/test_restore.py

```diff
@@ -0,0 +1,177 @@
+    os.environ["GITLAB_PROJECT"] = GITLAB_PROJECT
+
+    if ARTIFACTS_JOB:
+        os.environ["ARTIFACTS_JOB"] = ARTIFACTS_JOB
+
+    mock_os.path.join.side_effect = os.path.join
+    mock_os.getcwd.return_value = "fakeCWD"
+
+    runner = CliRunner()
+    result = runner.invoke(restore.main, ["pipeline_id"])
+
+    assert result.output == ""
+    assert result.exit_code == 0
+    assert result.exception is None
+
+    mock_gitlab_download.assert_called_once()
+    mock_gitlab_download.assert_called_with(
+        GITLAB_URL,
+        GITLAB_TOKEN,
+        GITLAB_PROJECT,
+        ARTIFACTS_JOB if ARTIFACTS_JOB else "backup",
+        "pipeline_id",
+        "fakeCWD/artifacts.zip"
+    )
+
+    mock_extract.assert_called_once()
+    mock_extract.assert_called_with("fakeCWD/artifacts.zip", "fakeCWD/artifacts")
+
+    mock_upload.assert_called_once()
+    mock_upload.assert_called_with("fakeCWD/artifacts")
+
+
+@pytest.mark.parametrize(
+    "pipeline_status, jop_status, artifact_job, artifact_name",
+    [
+        pytest.param("success", "success", "backup", "artifacts.zip", id="default"),
+        pytest.param("success", "success", "custom", "artifacts.zip", id="custom"),
+        pytest.param("abcdefg", "success", "backup", "artifacts.zip", id="pipeline status != 'success'"),
+        pytest.param("success", "abcdefg", "backup", "artifacts.zip", id="job status != 'success'"),
+        pytest.param("success", "success", "backup", "abcdefghijklm", id="artifact name != 'artifacts.zip'"),
+    ]
+)
+@mock.patch("atacac.restore.gitlab")
+def test_gitlab_download(mock_gitlab, pipeline_status, jop_status, artifact_job, artifact_name):
+    gl = mock_gitlab.Gitlab.return_value
+    project = mock.MagicMock()
+    pipeline = mock.MagicMock()
+    job = mock.MagicMock()
+    job_content = mock.MagicMock()
+
+    gl.projects.get.return_value = project
+
+    project.pipelines.get.return_value = pipeline
+    project.jobs.get.return_value = job_content
+
+    pipeline.id = 12345
+    pipeline.attributes = {
+        "status": pipeline_status,
+        "sha": "value_sha",
+        "web_url": "value_web_url"
+    }
+    pipeline.jobs.list.return_value = [job]
+
+    job.name = artifact_job
+    job.attributes = {
+        "status": jop_status,
+        "web_url": "value_web_url"
+    }
+    job.artifacts = [
+        {
+            "filename": artifact_name,
+            "file_format": "value_file_format",
+            "size": "value_size"
+        }
+    ]
+
+    with tempfile.NamedTemporaryFile() as file:
+        try:
+            restore.gitlab_download(
+                "VALUE_url",
+                "VALUE_token",
+                "VALUE_project",
+                artifact_job,
+                "VALUE_pipeline_id",
+                file.name
+            )
+        except Abort:
+            assert (pipeline_status != "success"
+                    or jop_status != "success"
+                    or artifact_name != "artifacts.zip")
```

Comment:
And if `restore.gitlab_download` wasn't raised Abort everthing is OK no matter what `restore.gitlab_download` did?